mlir/lib/Dialect/SPIRV/IR/SPIRVOps.cpp

```diff
@@ -1021,6 +1021,14 @@ LogicalResult spirv::FuncOp::verifyType() {
 
 LogicalResult spirv::FuncOp::verifyBody() {
   FunctionType fnType = getFunctionType();
+  if (!isExternal()) {
+    Block &entryBlock = front();
+
+    unsigned numArguments = this->getNumArguments();
+    if (entryBlock.getNumArguments() != numArguments)
+      return emitOpError("entry block must have ")
+             << numArguments << " arguments to match function signature";
+  }
```

Comment:
Can we also add the missing check for the type matching?

```
  spirv.func @f(f32) "None" {
    ^bb0(%arg0: f64):
    %c0 = arith.constant 0 : index
    spirv.Return
  }
```

Right now this is parsing correctly and passing the verifier, but it shouldn't.